---
title: "Lecture 15"
author: "Rachael McVicar"
date: "2/26/2020"
output: github_document
---
## Read sample genotypes data from ENSEMBL

We downloaded genotype data from ENSMEBLE on the MXL Mexican Ancestry in LA


```{r}
mxl<- read.csv("373531-SampleGenotypes-Homo_sapiens_Variation_Sample_rs8067378.csv")
head(mxl)
```

RM:I'm only interested in the Genotype. forward, strand
BG: We want to look at the second column that comtaains the genotype information

```{r}
table(mxl$Genotype..forward.strand.)
```

##RNA-Seq result analysis for the different genotypes of this SNP

```{r}
expr<- read.table("rs8067378_ENSG00000172057.6 (1).txt")
head(expr)
```

Does the expression change based on phenotype 
expr$geno, A/A, A/G, G/A, G/G

```{r}
expr$geno == "G/G"
```

for every true, I want to find the correlating expression value and find the average 
```{r}
summary( expr[expr$geno == "G/G" , ]$exp, breaks=30)
```

```{r}
hist( expr[expr$geno == "G/G" , ]$exp, breaks=20)
```

```{r}
summary( expr[expr$geno == "A/G" , ]$exp, breaks=20)
summary( expr[expr$geno == "G/G" , ]$exp, breaks=20)
summary( expr[expr$geno == "G/A" , ]$exp, breaks=20)
summary( expr[expr$geno == "A/A" , ]$exp, breaks=20)
```

##Try boxplot

We will use the `boxplot()` function and the input data will be the 
**expr**. HOw do we draw a useful boxplot?

```{r}
?boxplot
boxplot(exp ~ geno, data=expr, notch=TRUE)
```

Figuring out how many rows we are looking at here?
```{r}
nrow(expr)
```

# complement to the hands on session for lecture 13.
First we need to load our expression data from a file (that you downloaded):


```{r}
expr <- read.table("rs8067378_ENSG00000172057.6 (1).txt")
```

Or we can read from online.
```{r}
url <- "https://bioboot.github.io/bimm143_S18/class-material/rs8067378_ENSG00000172057.6.txt"
expr <- read.table(url)
```


Check the genotype sample size and overall summary stats 
```{r}
summary(expr)
```


Lets break it down to the three genotypes by 
```{r}
summary( expr[expr[,2] == "A/A",3] )
summary( expr[expr[,2] == "G/G",3] )
summary( expr[expr[,2] == "A/G",3] )
```


We could of also used `expr$geno == "A/A"` in the above, e.g.
```{r}
summary( expr$exp[ expr$geno == "A/A" ] )
```


Lets generate some nice plots of our results
Generate a boxplot for expression data of different genotype groups
```{r}
p <- boxplot(exp~geno, data=expr, xlab="rs8067378 genotype", ylab="ENSG00000172057.4 (RPKM)", notch=T)
```


Look at the output 'p' and you will see the median values etc. used to generate the plot
Optional: Using ggplot2

We could also use the **ggplot2** package to make our plots

```{r}
library(ggplot2)
```


Boxplot
```{r}
ggplot(expr, aes(geno, exp)) + geom_boxplot()
```


Histogram of the exp column with ggplot2
```{r}
ggplot(expr, aes(exp, fill = geno)) + geom_density(alpha = 0.2)
```


Boxplot with the data shown
```{r}
ggplot(expr, aes(geno, exp, fill=geno)) + 
  geom_boxplot(notch=TRUE, outlier.shape = NA) + 
  geom_jitter(shape=16, position=position_jitter(0.2), alpha=0.4)
```
