---
title: "Structural Bioinformatics (pt.1)"
author: "Rachael McVicar"
date: "2/14/2020"
output: github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
 
## PDB Statistics

Download a CSV file from the PDB site (accessible from “Analyze” > “PDB Statistics” > “by Experimental Method and Molecular Type”. Move this CSV file into your RStudio project and use it to answer the following questions:

> Q: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.

```{r}
stats <- read.csv("Data Export Summary.csv")
stats
#I want to change row names, so....
stats <- read.csv("Data Export Summary.csv", row.names = 1)
```

```{r}
stats[1, 1]/sum(stats$Proteins)
stats[1, 3]/sum(stats$Proteins)
(stats[1, 1]+stats[1,3])/sum(stats$Proteins)
```

Shoot, I want total structures, not just protein structures:b
```{r}
stats[1, 1]/sum(stats$Total) * 100
stats[1, 3]/sum(stats$Total) * 100
(stats[1, 1]+stats[1,3])/sum(stats$Total) * 100
```
I wanna make it purdy, so round it vvvv
```{r}
ans<- round(stats$Total/sum(stats$Total) * 100, 2)
names(ans)<- rownames(stats)
ans
```


> Q: What proportion of structures in the PDB are protein?
[1] 92.69057
I might have already started answering it, woops

```{r}
x_ray <- round(stats[1, 1]/sum(stats$Proteins) * 100, 2)
electron_microscopystats <- round(stats[1, 3]/sum(stats$Proteins) * 100, 2)
x_ray
electron_microscopystats
```
just realized I didn't answer the question, yikes
```{r}
round(sum(stats$Proteins)/sum(stats$Total) * 100, 2)
```

> Q: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?
This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

##Using bio3d to analyse protein structure data

Use the bio3d package from CRAN, it's already installed
```{r}
library(bio3d)
pdb <- read.pdb("1hsg")
pdb
```

I wanna make it lung based, let's bring in Dr.Leibel's favorite molecule
SURFACTANT PROTEIN WOOOO (D instead of B though)
```{r}
library(bio3d)
pdb <- read.pdb("1ake")
pdb
attributes(pdb)
head(pdb$atom)
```

> Q: How many amino acid residues are there in this pdb object? 
11412

> Q: Name one of the two non-protein residues? 
[ AP5 (2), HOH (378) ]

> Q: How many protein chains are in this structure? 
Chains#: 2  (values: A B)

```{r}
library(bio3d)
pdb <- read.pdb("1ake")
pdb
```

Adenylate kinase structures from the PDB
Search and retrieve ADK structures

```{r}
hits <- NULL
hits$pdb.id <- c('1AKE_A','4X8M_A','6S36_A','6RZE_A','4X8H_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','4NP6_A','3GMT_A','4PZL_A')
```

Download and parse these structures
```{r}
#Download related PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```

Align and superpose structures
```{r}
# Align releated PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
```

```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
plot(pdbs, labels=ids)
```

Viewing these superposed structures
We can view our superposed results with the new bio3d.view view() function:

##library(bio3d.view)
##library(rgl)
##view(pdbs)
I can't view it, I give up, I feel like I'm going around in circles

```{r}
pc<- pca(pdbs)
plot(pc)
```

# Perform PCA
```{r}
pc.xray <- pca(pdbs)
plot(pc.xray)
```

```{r}
plot(pc, 1:2)
dim(pc$U)
```

# Calculate RMSD
```{r}
rd <- rmsd(pdbs)
```


# Structure-based clustering
```{r}
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
```


# Visualize first principal component
```{r}
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
```

```{r}
View(pc1)
```

```{r}
#res <- read.pdb("all.pdbqt", multi=TRUE)
#write.pdb(res, "results.pdb")
```

